src/main/java/org/folio/services/holding/HoldingsService.java

```diff
@@ -127,30 +143,34 @@ public Future<Response> deleteHoldings(String cql) {
   }
 
   public Future<Response> createHoldings(List<HoldingsRecord> holdings, boolean upsert, boolean optimisticLocking) {
-    if (upsert) {
-      return upsertHoldings(holdings, optimisticLocking);
-    }
-
     for (HoldingsRecord holdingsRecord : holdings) {
```

Comment:
In the previous version of the code, effective location is not set when upsert is involved. In the current version, effective location is always set. Will this break something?

Reply:
It was set inside of upsertHoldings: https://github.com/folio-org/mod-inventory-storage/pull/930/files#diff-78dff0242682c91208ab3b3ce644115ee6d3cb522c0697639db65927241547feL150
Now it's just set in a single place for both upsert and non upsert

Reply:
Ah. good catch!

Reply:
Actually a thought occurred to me. upsertHoldings is a public method so it may be called by itself. If it is called by itself, effective location will not be set. What do you think?

Reply:
Good point. As I can see upsertHoldings  was added recently and is called only from createHoldings for now. Not sure if there were plans to use it somewhere else. Probably it could be just made private? 

Or if it should remain public, then it should contain not only setting effective locations, but also creating shadow instances, as in createHoldings, right?  

@julianladisch , what are your thoughts on this?

Reply:
the name upsertHoldings tells me that holdings can be created as well as updated. I would be okay to leave as is if only updates were done; that means the holding is already under an instance. So lets keep the original intent of author of this method and keep it public and add setting effective locations and creating shadow instances.